feat(rq): add section on task queues (#83)
tecladocode/rest-apis-flask-python: docs/docs/12_task_queues_emails/04_populate_rq_task_queue/README.md

The background worker will import `tasks.py` once at the start of its lifetime, so doing this will (very slightly) improve performance.

We could leave it like this, but I think we can do better. Let's write another function underneath that one that specifically describes the task that we want to perform in the background: send a registration email to a specific user:

```py title="tasks.py"
import os
import requests
from dotenv import load_dotenv

load_dotenv()

DOMAIN = os.getenv("MAILGUN_DOMAIN")

def send_simple_message(to, subject, body):
    return requests.post(
        f"https://api.mailgun.net/v3/{DOMAIN}/messages",
        auth=("api", os.getenv("MAILGUN_API_KEY")),
        data={"from": f"Your Name <mailgun@{DOMAIN}>",
            "to": [to],
            "subject": subject,
            "text": body}
    )

# highlight-start
def send_user_registration_email(email, username):
    return send_simple_message(
        email,
        "Successfully signed up",
        f"Hi {username}! You have successfully signed up to the Stores REST API.",
    )
# highlight-end
```

:::tip
Remember to change "Your Name" in `from` to whatever name you want your emails to come from!
:::

> Next up, let's get our Redis connection string and add it to the `.env` file:

Reviewer: ```suggestion
Next up, add the Redis connection string that we got in the [previous section](add permalink) to the `.env` file:
```

Note, update the link to the previous section.

```text title=".env"

> REDIS_URL="rediss://red-ct8aen0hkl10:MnLs0mmrX7MBXWRkdrh49@frankfurt-redis.render.com:6379"

Reviewer: ```suggestion
REDIS_URL="<insert your Redis url here>"
```

Author: Nice one, done!
